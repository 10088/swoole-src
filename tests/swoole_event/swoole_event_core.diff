001+ Fatal error: Uncaught Error: Call to undefined function pcntl_fork() in /src/swoole-src/tests/swoole_event/swoole_event_core.php:12
001- SUCCESS
002+ Stack trace:
003+ #0 {main}
004+   thrown in /src/swoole-src/tests/swoole_event/swoole_event_core.php on line 12