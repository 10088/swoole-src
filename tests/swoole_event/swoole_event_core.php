
<?php
function dnsLookup() {
    swoole_async_dns_lookup("www.qq.com", function($host, $ip) {
        swoole_event_exit();
        exit();
    });
}

$i = 200;
while (--$i) {
    $pid = pcntl_fork();
    if ($pid < 0) {
        exit;
    }

    if ($pid === 0) {
        dnsLookup();
        exit();
    }

    pcntl_waitpid($pid, $status);
    if (!pcntl_wifexited($status)) {
        fprintf(STDERR, "$pid %s exit [exit_status=%d, stop_sig=%d, term_sig=%d]\n",
            pcntl_wifexited($status) ? "normal": "abnormal",
            pcntl_wexitstatus($status),
            pcntl_wstopsig($status),
            pcntl_wtermsig($status)
        );
        exit(1);
    }
}
echo "SUCCESS";
?>

