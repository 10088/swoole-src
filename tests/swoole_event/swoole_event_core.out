Fatal error: Uncaught Error: Call to undefined function pcntl_fork() in /src/swoole-src/tests/swoole_event/swoole_event_core.php:12
Stack trace:
#0 {main}
  thrown in /src/swoole-src/tests/swoole_event/swoole_event_core.php on line 12